src/components/Kv/KvLightbox.vue

```diff
@@ -0,0 +1,217 @@
+<template>
+	<transition name="fade">
+		<div v-show="isShown"
+			class="kv-lightbox-wrap"
+			ref="kvlightbox"
+			@keyup.esc="closeLightbox"
+			tabindex="500">
+			<div class="kv-lightbox-bg" @click.stop.prevent="closeLightbox"></div>
+			<div class="kv-lightbox">
+				<div class="row">
+					<div class="columns">
+						<div class="lightbox-content" :class="removeContentBg">
+							<button @click.stop.prevent="closeLightbox" class="close-lightbox" aria-label="Close">
+								<kv-icon name="x" />
+							</button>
+							<slot name="title"></slot>
+							<slot>Lightbox content</slot>
+							<br>
+							<slot name="controls">
+								<kv-button v-if="showCloseButton" @click.native="closeLightbox">
+									Close
+								</kv-button>
+							</slot>
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+	</transition>
+</template>
+
+<script>
+import KvIcon from '@/components/Kv/KvIcon';
+import KvButton from '@/components/Kv/KvButton';
+
+export default {
+	components: {
+		KvIcon,
+		KvButton
+	},
+	data() {
+		return {
+			isShown: false
+		};
+	},
+	props: {
+		visible: {
+			type: Boolean,
+			default: false
+		},
+		inverted: {
+			type: Boolean,
+			default: false
+		},
+		showCloseButton: {
+			type: Boolean,
+			default: false
+		}
+	},
+	computed: {
+		removeContentBg() {
+			return this.inverted ? 'inverted' : '';
+		}
+	},
+	watch: {
+		// Create and set our internal visibility property
+		visible() {
+			this.isShown = this.visible;
+			// if the lightbox is shown give it focus
+			// this ensures the esc functionality works
+			if (this.isShown) {
+				this.$nextTick(() => {
+					this.$refs.kvlightbox.focus();
+					this.lockScroll();
+				});
+			}
+		}
+	},
+	methods: {
+		closeLightbox() {
+			this.isShown = false;
+			// listen for this event in parent components
+			// it gives notice of the lightbox being closed internally
+			this.$emit('lightbox-closed');
+			// remove scroll lock class from body
+			this.unlockScroll();
+		},
+		lockScroll() {
+			if (typeof window !== 'undefined') {
+				document.body.classList.add('scroll-locked');
+			}
+		},
+		unlockScroll() {
+			if (typeof window !== 'undefined') {
+				document.body.classList.remove('scroll-locked');
+			}
+		}
+	}
+};
+</script>
+
+<style lang="scss">
+@import 'settings';
+
+.fade-enter-active,
+.fade-leave-active {
+	transition: opacity 0.5s;
+}
+
+.fade-enter,
+.fade-leave-to {
+	opacity: 0;
+}
```

Comment:
it's not a problem here, but these rules are overriding the same rules in TheTipMessage.vue

Reply:
I thought of that as well and think we should probably create a "transitions" scss file we can import that has our global styles for different transitions... timing, tweens, etc...

Reply:
I like that idea